lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -1596,7 +1596,7 @@ def customize_vm(self, vm_obj):
             # Setting hostName, orgName and fullName is mandatory, so we set some default when missing
             ident.userData.computerName = vim.vm.customization.FixedName()
             # computer name will be truncated to 15 characters if using VM name
-            default_name = self.params['name'].translate(None, string.punctuation)
+            default_name = self.params['name'].translate(str.maketrans('', '', string.punctuation))
```

Comment:
Thank you @wilmardo for your contribution. If I'm not mistaking, this won't work with Python2. Its`str` class does not have a `maketrans` method:

```
Python 2.7.15 (default, Oct 15 2018, 15:26:09) 
[GCC 8.2.1 20180801 (Red Hat 8.2.1-2)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import string
>>> 'aaaaaa?##a'.translate(str.maketrans('', '', string.punctuation))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: type object 'str' has no attribute 'maketrans'
```

Reply:
@goneri Hmm thought I tested this, thanks for the catch!
Now tested and fixed with re.sub, somewhat less efficient but works nicely with Python2 and Python3